src/services/chat_service.py

```diff
@@ -1,179 +1,146 @@
-                    return f"For {mission_id}: {self.mission_advice[mission_id]}"
-        
-        # Context-based responses
-        if context.page in self.context_responses:
-            base_responses = self.context_responses[context.page]
-            base_reply = random.choice(base_responses)
-        else:
-            base_reply = "I'm here to help you with your AI agent project. What would you like to know?"
-        
-        # Message-specific modifications
-        if any(word in message for word in ["mission", "next", "what should i do"]):
-            if session_data.missions:
-                pending_missions = [m for m in session_data.missions if m.id not in session_data.completed_missions]
-                if pending_missions:
-                    next_mission = pending_missions[0]
-                    return f"I'd recommend focusing on '{next_mission.title}' next. {self.mission_advice.get(next_mission.id, 'This will help advance your AI agent project.')}"
-            return "Let's start by defining your success metrics. This foundation will guide all other decisions."
-        
-        elif any(word in message for word in ["progress", "points", "completed"]):
-            return f"You've earned {session_data.points_total} points so far and completed {len(session_data.completed_missions)} missions. {"Your call is now unlocked!" if session_data.is_call_unlocked() else "Complete 2 missions to unlock your free consultation call."}"
-        
-        elif any(word in message for word in ["case study", "example", "learn"]):
-            return "Case studies are invaluable for understanding real-world applications. I can recommend specific ones based on your goals and current progress."
+        messages.append(SystemMessage(content=sys_prompt))
+
+
+        try:
+            llm_response = await llm.ainvoke(messages)
+            response = llm_response.content
+            # await history.aadd_messages(messages=[SystemMessage(content=sys_prompt), AIMessage(content=response)])
+            return response
+        except Exception as e:
+            print(f"LLM Parse Error: {e}", traceback.format_exc())
+
+
+
         
-        elif any(word in message for word in ["stuck", "confused", "help"]):
-            return "No worries! Building an AI agent can seem overwhelming. Let's break it down into manageable steps. What specific area would you like clarification on?"
+
+
+    async def ask(self, *, session_id: str, message: str, page: str, section: str):
+        """Generate a response to an ask message"""
+        progress = await session_manager.get_session_progress(session_id)
+        history = await get_history(session_id)
+        messages = await history.aget_messages()
+
+        if (len(messages) <= 1):
+            raise_http_error(404, "Session does not have a goal")
+
+        if (len(messages) == 5):
+            print("Adding System prompt")
```

Comment:
update the log to "Adding follow up system prompt"

Reply:
Fixed